imhamt/src/hamt.rs

```diff
@@ -130,6 +130,7 @@ impl<H: Hasher + Default, K: Eq + Hash + Clone, V: Clone> Hamt<H, K, V> {
     pub fn update<F, U>(&self, k: &K, f: F) -> Result<Self, UpdateError<U>>
     where
         F: FnOnce(&V) -> Result<Option<V>, U>,
+        U: std::error::Error + std::fmt::Debug + 'static,
```

Comment:
Could add use statements at the top rather than absolute paths everywhere.

Reply:
Why are these extra bounds needed here? They don't look essential to these methods.

Reply:
`U` is the `#[source]` of an error variant inside of `UpdateError`. To be the `#[source]` an error must be at least `Error + 'static` (see [`Error::source`](https://doc.rust-lang.org/std/error/trait.Error.html#method.source)). Since `UpdateError` is `Debug`, `U` must also be `Debug`.

Reply:
Ah, I see.